src/app/workflow/view/view.component.ts

```diff
@@ -119,26 +125,56 @@ export class ViewWorkflowComponent extends View implements OnInit {
       }
     );
   }
+
+  /**
+   * Handles the create DOI button being clicked
+   *
+   * @memberof ViewWorkflowComponent
+   */
+  requestDOIForWorkflowVersion() {
+    this.doiRequested = true;
+    this.viewService.requestDOIForWorkflowVersion(this.workflow, this.version, () => {
+      this.doiRequested = false;
+    });
+  }
+
   /**
    * Opens a confirmation dialog that the Dockstore User can use to
    * confirm they want a snapshot.
    *
    * @memberof ViewWorkflowComponent
    */
   snapshotVersion(): void {
```

Comment:
Move the bulk of this function into a service.  If you really want to keep this.snapshotRequested, just move that to the state